code/game/mecha/equipment/tools/tools.dm

```diff
@@ -363,13 +363,44 @@
 	energy_drain = 1000
 	range = RANGED
 
-/obj/item/mecha_parts/mecha_equipment/teleporter/action(atom/target)
+/obj/item/mecha_parts/mecha_equipment/teleporter/action(atom/target, mob/user)

```

Comment:
The mecha should be using do_teleport. What exactly does this do that do_teleport doesn't?

Reply:
do_teleport I believe is registering the mech when it moves as a impediment and is creating large emissions even if you teleport onto a safe tile. Other part is I wanted abit more assorted movement then just moving it to a safe tile, IE more randomized.

Reply:
Then fix do_teleport, don't create snowflake checks in the teleporter. Regarding the second point, I am fairly certain do_teleport already does that. Ultimately regardless of what you do you should be using do_teleport even if you perform some checks beforehand (such as inducing halloss to the pilot) because do_teleport will handle both rangefinding, non-turf impediments, and is recursive.